packages/swapper/src/swappers/ChainflipSwapper/getChainflipTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,282 @@
+        // TODO: Solana gas calc
+        return undefined;
+      }
+    }
+    
+    return undefined;
+  }
+  
+  const getFeeAsset = (fee: ChainflipBaasQuoteQuoteFee) => {
+    if (fee.type === "ingress" || fee.type === "boost")
+      return sellAsset;
+
+    if (fee.type === "egress")
+      return buyAsset;
+
+    if (fee.type === "liquidity" && fee.asset == sellChainflipChainKey)
+      return sellAsset;
+
+    if (fee.type === "liquidity" && fee.asset == buyChainflipChainKey)
+      return buyAsset;
+
+    if (fee.type === "liquidity" && fee.asset == "usdc.eth")
+      return usdcAsset;
+
+    if (fee.type === "network")
+      return usdcAsset;
+  }
+  
+  const getProtocolFees = (singleQuoteResponse: ChainflipBaasQuoteQuote) => {
+    const protocolFees: Record<AssetId, ProtocolFee> = {}
+
+    for (const fee of singleQuoteResponse.includedFees!) {
+      if (fee.type === "broker")
+        continue;
+
+      const asset = getFeeAsset(fee)!
+      if (!(asset.assetId in protocolFees)) {
+        protocolFees[asset.assetId] = {
+          amountCryptoBaseUnit: "0",
+          requiresBalance: false,
+          asset: asset,
+        }
+      }
+ 
+      protocolFees[asset.assetId].amountCryptoBaseUnit =
+        (BigInt(protocolFees[asset.assetId].amountCryptoBaseUnit) + BigInt(fee.amountNative!)).toString();
+    }
+
+    return protocolFees;
+  }
+  
+  const quotes = [];
+  
+  for (const singleQuoteResponse of quoteResponse) {
+    const isStreaming = singleQuoteResponse.type === "dca";
+        
+    if (singleQuoteResponse.boostQuote) {
+      const boostRate = getRate({
+        sellAmountCryptoBaseUnit: singleQuoteResponse.boostQuote.ingressAmountNative!,
+        buyAmountCryptoBaseUnit: singleQuoteResponse.boostQuote.egressAmountNative!,
+        sellAsset,
+        buyAsset,
+      })
+      
+      const boostSwapSource = singleQuoteResponse.type === "regular"
+        ? CHAINFLIP_BOOST_SWAP_SOURCE
+        : CHAINFLIP_DCA_BOOST_SWAP_SOURCE;
+      
+      const boostTradeQuote: TradeQuote = {
+        id: uuid(),
+        rate: boostRate,
+        receiveAddress: "",
+        potentialAffiliateBps: commissionBps,
+        affiliateBps: commissionBps,
+        isStreaming: isStreaming,
+        slippageTolerancePercentageDecimal: defaultSlippage,
+        steps: [
+          {
+            buyAmountBeforeFeesCryptoBaseUnit: singleQuoteResponse.boostQuote.egressAmountNative!,
+            buyAmountAfterFeesCryptoBaseUnit: singleQuoteResponse.boostQuote.egressAmountNative!,
+            sellAmountIncludingProtocolFeesCryptoBaseUnit: singleQuoteResponse.boostQuote.ingressAmountNative!,
+            feeData: {
+              networkFeeCryptoBaseUnit: await getGasFee(),
+              protocolFees: getProtocolFees(singleQuoteResponse.boostQuote),
+            },
+            rate: boostRate,
+            source: boostSwapSource,
+            buyAsset: buyAsset,
+            sellAsset: sellAsset,
+            accountNumber: 0,
```

Comment:
Note this is very dangerous and we should ensure we pass the `accountNumber` (account_index part from BIP-44 derivation path spec) or this will get a quote tied to, and eventually executed with account 0 even if users selected another account of theirs

Reply:
This is way above my understanding ;) I have no idea what an `accountNumber` is. Could you help me out what I should be using?

Reply:
Sure thing! I like this link a lot and I always send it out when the question of derivation paths comes in: https://support.bifrostwallet.com/en/articles/5588263-derivation-paths-explained

> As an example, Ledger hardware wallets increase the account value, meaning that later addresses are derived using

This is the way we increment in the BIP44 spec too - and probably the way all wallets should have done it too, which would have saved users a lot of headaches with different derivation paths across wallets ;)

ShapeShift supports multiple account numbers "connected" at the same time (in quotes, because for the wallets that *do* support multiple account numbers i.e KeepKey, Ledger, and ShapeShift native, there really isn't a notion of connected account, they're just all here at the same time ready to be used.

An image is worth a thousand words so for users that means something like this:

<img width="1377" alt="image" src="https://github.com/user-attachments/assets/25516277-6139-4e90-834e-ee19ee1fa0ee">
<img width="1382" alt="image" src="https://github.com/user-attachments/assets/8a8661e0-42c6-4f0e-9c70-1dc842f8c094">
<img width="639" alt="image" src="https://github.com/user-attachments/assets/e6bbf4ce-305a-4545-989f-bbce38208863">

Reply:
I found `accountNumber` in the `input: GetTradeQuoteInput` but there it can be undefined, what do you pass in that case?

Reply:
Let's early return if undefined for the time being @CumpsD 🙏🏽 !  The refactor in progress will soon split between rates and quotes, but as far as this PR is concerned, assume what you're getting here at input time is going to be used e2e all the way until execution.